### PR TITLE
Suite sparse compilation fix for 5.7.0 and 5.7.1

### PR DESCRIPTION
**Pull request by mathsen**
Newest suite-sparse versions also need separate make run to compile successfully

### CODE REVIEW

**Comment by adamjstewart** [comment]
It looks like you accidentally incorporated the changes from another PR into this one.

**Comment by mathsen** [comment]
You are absolutely right, I mistakenly included a commit from [PR 15288](https://github.com/spack/spack/pull/15288) and dropped it now.

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 1]
```diff
@@ -131,8 +131,7 @@ def install(self, spec, prefix):
         # In those SuiteSparse versions calling "make install" in one go is
         # not possible, mainly because of GraphBLAS.  Thus compile first and
         # install in a second run.
-        if (self.spec.version >= Version('5.4.0') and
-            self.spec.version <= Version('5.6.0')):
+        if (self.spec.version >= Version('5.4.0')):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: Can we change it with the more idiomatic:
```python
if '@5.4.0:' in self.spec:
```
?

Comment (mathsen) [reply var/spack/repos/builtin/packages/suite-sparse/package.py]: Thanks for the hint and done :)

**Comment by alalazo** [comment]
Thanks!

**Pull request merged**